src/executor/plan_executor.cpp

```diff
@@ -82,7 +82,19 @@ ExecuteResult PlanExecutor::ExecutePlan(const planner::AbstractPlan *plan,
 
       // Execute the tree until we get result tiles from root node
       while (status == true) {
-        status = executor_tree->Execute();
+        // Handle constraint exceptions
+        // TODO: This is a temporary solution
+        // The best way to handle constraint violations is NOT using
+        // exceptions
+        try {
+          status = executor_tree->Execute();
+        } catch (Exception &e) {
+          p_status.m_processed = executor_context->num_processed;
+          p_status.m_result = ResultType::FAILURE;
+          p_status.m_result_slots = nullptr;
+          CleanExecutorTree(executor_tree.get());
+          throw e;
```

Comment:
Yes, these ConstraintExceptions are caught in the traffic cop, which then updates the transaction result appropriately.